libtiny_tui/src/msg_area/line.rs

```diff
@@ -1,34 +1,29 @@
+use crate::{
+    config, config::Colors, line_split::LineDataCache, utils::translate_irc_control_chars,
+};
 use std::mem;
+use std::ops::{Deref, DerefMut};
 use termbox_simple::{self, Termbox};
 
-use crate::{config, config::Colors, utils::translate_irc_control_chars};
-
 /// A single line added to the widget. May be rendered as multiple lines on the
 /// screen.
 #[derive(Debug)]
 pub(crate) struct Line {
     /// Line segments.
-    segs: Vec<Seg>,
+    string_segments: Vec<StyledString>,
 
     /// The segment we're currently extending.
-    current_seg: Seg,
+    current_seg: StyledString,
 
     /// Number of characters in the line (includes all segments).
     len_chars: i32,
 
-    /// Char indices (NOT byte indices!) of split positions of the line - when
-    /// the line doesn't fit into the screen we split it into multiple lines
-    /// using these positions as split points.
-    ///
-    /// It's important that these are really indices ignoring invisible chars,
-    /// as we use difference between two indices in this vector as length of
-    /// substrings.
-    splits: Vec<i32>,
+    line_data: LineDataCache,
 }
 
 #[derive(Debug)]
-struct Seg {
-    text: String,
+struct StyledString {
+    inner: String,
```

Comment:
IME the `inner` naming convention is mostly used when you wrap a struct with reference types and give it a new name, e.g. we have

```rust
pub struct TUI {
    inner: Weak<RefCell<tui::TUI>>,
}
```

For this type I think it's better to just call it `str` or `string`.